gpytorch/kernels/kernel.py

```diff
@@ -194,11 +201,8 @@ def _set_lengthscale(self, value):
         self.initialize(raw_lengthscale=self._inv_param_transform(value))
 
     def size(self, x1, x2):
-        non_batch_size = (x1.size(-2), x2.size(-2))
-        if x1.ndimension() == 3:
-            return torch.Size((x1.size(0),) + non_batch_size)
-        else:
-            return torch.Size(non_batch_size)
+        non_batch_shape = torch.Size([x1.size(-2), x2.size(-2)])
+        return x1.shape[:-2] + non_batch_shape
```

Comment:
Just a thought - should we support broadcasting for kernel params?

For example - imagine you want to pass `n x d` data through kernels with `b` different lengthscales. You could construct a kernel with `batch_shape=torch.Size((b,))` and then pass `x` through it to get an output of size `b x n x n`.

Reply:
In that case, you'd probably want to determine the batch shape using one of the broadcast helpers.

Reply:
Actually, you probably should determine the batch shape using the broadcast helper anyways, in case x1 and x2 have different batch shapes?